pkg/controllers/dynakube/extension/secret.go

```diff
@@ -0,0 +1,63 @@
+package extension
+
+import (
+	"context"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta3/dynakube"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/dttoken"
+	k8ssecret "github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/secret"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	eecTokenKey         = "eec-token"
+	eecTokenValuePrefix = "EEC dt0x01"
+	secretSuffix        = "-extensions-token"
+)
+
+func reconcileSecret(ctx context.Context, dk *dynakube.DynaKube, kubeClient client.Client, apiReader client.Reader) error {
+	log.Info("reconciling secret")
+
+	query := k8ssecret.NewQuery(ctx, kubeClient, apiReader, log)
+
+	_, err := query.Get(client.ObjectKey{Name: getSecretName(dk.Name), Namespace: dk.Namespace})
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+
+	if errors.IsNotFound(err) {
+		log.Info("creating secret")
+
+		newEecToken, err := newEecToken()
```

Comment:
```suggestion
		newEecToken, err := dttoken.New(eecTokenValuePrefix)
```
just not sure that we need new function for 1 line 🤷

Reply:
[Fixed](https://github.com/Dynatrace/dynatrace-operator/pull/3426/commits/6bc7372468253d455adb611f2097fb5033f7a012)